### PR TITLE
DEPR: Passing in a string column label for DataFrame.ewm(times=...)

### PR DESCRIPTION
**Pull request by mroeschke**
- [x] xref https://github.com/pandas-dev/pandas/pull/42834#discussion_r680451102
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [x] whatsnew entry

### CODE REVIEW

**Comment by jreback** [comment]
umm we generally accept a column name for other things, why are we deprecating here?

**Comment by mroeschke** [comment]
- https://github.com/pandas-dev/pandas/pull/42834 deprecated dropping nuisance column in window ops (raising a `TypeError` in the future)
- A `times` column will always be a nuisance column because windows ops only accept numeric values and `times` will be a `datetime64[ns]`, so this functionality as is will always raise a `TypeError` in the future

File: pandas/tests/window/test_ewm.py [hunk 1]
```diff
@@ -231,3 +230,14 @@ def test_float_dtype_ewma(func, expected, float_numpy_dtype):
...
 
     tm.assert_frame_equal(result, expected)
+
+
+def test_times_string_col_deprecated():
+    # GH 43265
+    data = np.arange(10.0)
+    data[::2] = np.nan
+    df = DataFrame({"A": data, "time_col": date_range("2000", freq="D", periods=10)})
+    with tm.assert_produces_warning(FutureWarning, match="Specifying times"):
```
Comment (jreback) [inline pandas/tests/window/test_ewm.py:240]: do we have any tests with times as an array (you do above in test_ewma_with_times_equal_spacing but you are also hitting the nuiscance path

Comment (mroeschke) [reply pandas/tests/window/test_ewm.py:240]: This test passes `times` as an array


https://github.com/pandas-dev/pandas/blob/a6943ae49f79452da6ca5f1c3e008ded4e0e2c85/pandas/tests/window/test_ewm.py#L126-L136

**Comment by jreback** [comment]
thanks @mroeschke

**Pull request merged**